internal/common/types.go

```diff
@@ -0,0 +1,105 @@
+// Package common defines types to be used throughout the osbuild-composer, osbuild-worker,
+// and any other possible future executables. These types should also be used to translate
+// a user input into more-type safe representation and then back to a user-facing output.
+package common
+
+import (
+	"github.com/google/uuid"
+)
+
+// ParseCustomTypeError is thrown when parsing strings into enumerations
+type ParseCustomTypeError struct {
+	reason string
+}
+
+func (err *ParseCustomTypeError) Error() string {
+	return err.reason
+}
+
+// Architecture represents one of the supported CPU architectures available for images
+// produced by osbuild-composer. I decided to represent it as an integer because if it
+// was a string it would unmarshal from JSON just fine even in case that the architecture
+// was unknown (yeah, thanks Go for doing the same mistakes C did almost 50 years ago)
+//
+// I considered some alternatives like a struct with private fields, but struct cannot
+// be const, so that does not help either. Eventually I think this is the "get s**t done"
+// solution and next time please use a language with decent type system.
+type Architecture int
+
+// A list of supported architectures. As the comment above suggests the type system does
+// not allow to create a type with a custom set of values, so it is possible to use e.g.
+// 56 instead of an architecture, but as opposed to a string it should be obvious that
+// hardcoding a number instead of an architecture is just wrong.
+const (
+	X86_64 Architecture = iota
+	Aarch64
+	Armv7hl
+	I686
+	Ppc64le
+	S390x
+)
+
+// UnmarshalJSON is a custom unmarshaling function to limit the set of allowed values
+// in case the input is JSON.
+func (arch *Architecture) UnmarshalJSON(data []byte) error {
+	archString := string(data)
+	switch archString {
+	case `"x86_64"`:
+		*arch = X86_64
+	case `"aarch64"`:
+		*arch = Aarch64
+	case `"armv7hl"`:
+		*arch = Armv7hl
+	case `"i686"`:
+		*arch = I686
+	case `"ppc64le"`:
+		*arch = Ppc64le
+	case `"s390x"`:
+		*arch = S390x
+	default:
+		return &ParseCustomTypeError{archString + " is not a valid architecture"}
+	}
+	return nil
+}
+
+type ImageType int
+
+const (
```

Comment:
I think this and Distribution should also have unmarshal functions, depending on int values for the inputs is going to be fragile and difficult to keep synchrnized with the front ends when they are extended.

Reply:
Sure, I'll implement Unmarshal as well. Would you suggest to use something else than `int`?

Reply:
I think using int constants internally is fine.